WIP - Assorted torch related bug fixes and improvements

atlantiscze opened the pull request:
Commit 1:
- Adds new "Vertical Conduit" machine that acts as new replacement to multiZ cables. Cheap to build. Old cables cannot be laid (though can be mapped in) and no longer work (they didn't work most of the time anyway)
- Adds bunch of new SMES subtypes, updates some existing ones. Will be used by following commit(s)
- Rips out a piece of my bluespace drives code that fixes shuttle power interaction and generally makes shuttle power related checks less laggy.
- Fixes #15558    (in combination with next commits that map in the new variant)

Commit 2:
- Updates torch maps. See commit for list of changes

TODO: Commit 3: Update exodus map.

atlantiscze committed:
Assorted torch related bug fixes and improvements - Stage 1 - Code changes
- Adds new "Vertical Conduit" machine that acts as new replacement to multiZ cables. Cheap to build. Old cables cannot be laid (though can be mapped in) and no longer work (they didn't work most of the time anyway)
- Adds bunch of new SMES subtypes, updates some existing ones. Will be used by following commit(s)
- Rips out a piece of my bluespace drives code that fixes shuttle power interaction and generally makes shuttle power related checks less laggy.

atlantiscze committed:
Makes Z conduits and relevant boards (and breaker boxes as a bonus) buildable.
- They are generally cheap.
- Few other remaining minor code changes.

atlantiscze committed:
Stage 2 - Torch Map Changes
- This is a pretty large map change commit
- Replaces old Z conduits with new one. There's a large Z1-Z4 conduit near engineering, Z4-Z5 is separate.
- Fixes a ton of missing areas, therefore fixing shield tile acquisition. Whole torch should be fully shieldable now.
- Upgraded substations, they now have 2x larger capacity. They also house shield generators now, dedicated shield rooms were removed. Shields are connected to the main grid
- Enhanced engine SMES, it's still not ideal, but it's generally better and easier to upgrade given availability of coils on the torch.
- Rewired hangar deck, each shuttle now has it's own SMES as power storage, charged from main hangar SMES. These have pretty high IO rate so they charge very quickly, and are already ready for eventual addition of bluespace drives.
- Engine filters now start turned off, as they are on exodus, for consistency. Engine radiator changed by few tiles, overall cooling performance should be roughly the same.
- Engine TEG cold loop is now connected in parallel, as it is on the exodus.
- Adds few cameras at hangar deck and bridge solar control

Hopefully i didn't forget anything.

comma commented:
So now if an engineer wants to lay vertical cable, instead of using cable coil they need to build a machine with metal and a circuit board?

MistakeNot4892 commented:
Why not just fix the cables?

atlantiscze commented:
Because the old vertical cable code is totally disgusting.

The machine is cheap, and even multiple of those can be built with resources engineering has on hand. I forgot to map in spare boards, but i'll rectify that.

BlueNexus commented:
I don't think it should require a circuit board, honestly.

atlantiscze commented:
If i want to use frame-based construction system, it needs a circuit board. Other than that, just a bunch of wires.

Also, over old cables, this has an advantage of allowing you to stack these on top of each other, to create one large multiZ conduit pretty easily.

I admit, the sprite isn't ideal (reused breaker box sprite, description has been written to say it's a wiring closet so it at least roughly matches the sprite), but if anyone who is better in spriting wants to make it a dedicated sprite, i have absolutely no objections.

comma commented:
It's not about being expensive resource wise it's about being a pain compared to just having a coil of wire on you. Hardly and improvement

comma commented:
woops wrong button

atlantiscze commented:
As i said, the original Z cables were quite ugly mess. If someone wants to implement an alternative solution, why not, but i am not planning to waste time trying to fix them when i've already created a fully functional (and in some aspects even superior, such as the ability to make a Z cable spanning 3 or more Z levels at once) replacement.

EDIT: The construction difficulty is comparable to an air alarm, and actually lower than an APC. Tell me, how often do you have to actually build a vertical cable? Right now, perhaps once per round (unfucking current bugged cables by cut & replace method doesn't count)

BlueNexus commented:
I really don't like having to use a circuit board for this. Change a few things so you don't need a circuit board for it? Maybe make the frame constructable with steel, as a new option, and then they add wire.

mwerezak commented:
On balance, requiring a circuit board doesn't seem like all that much of a big deal IMO. I'd still be nice to actually solve the multi-z cable problem though.

atlantiscze commented:
It solves the problem, by replacing multiZ cables altogether.

comma commented:
Cable to machinery is nor a proper fix nor an improval. 👎  on that part.

comma reviewed (changes requested):
Either properly fix cables or remove your conduit hack.

atlantiscze commented:
My solution resolves the problem, with added benefit for allowing for easy stacking through multiple Zs.

Since the powernet bug is probably the most severe one i'm fixing with this, i am not removing that. If you don't like this solution, fix it yourself.

mwerezak commented:
@atlantiscze since people are against the idea of a machine, perhaps you could make the zcable obj be created by placing cables onto open space?

atlantiscze commented:
@mwerezak I'll do that.

comma commented:
Sure. Can you explain exactly what was the issue with cables? I am unable to reproduce it on example map, both up-down and down-up connections power properly.

![](http://i.imgur.com/WkS6OJV.png)
![](http://i.imgur.com/QBiZp14.png)
is the setup I have

EDIT: depowered areas & charges SMESes on Torch with admin secrets, all decks seem to be properly connected.

Can you give your setup, where you could see issue appearing / being fixed?

atlantiscze commented:
The issue is commonly occuring on the torch, especially when you start playing with things that update powernets - toggling breakers, cutting and replacing cables, etc.

I also can't dismiss the possibility that current implementation of shuttle power handling (which was originally designed as a bandaid fix for engineering outpost losing power, it allowed for a SMES to be used to transport energy from station in a shuttle). APCs in shuttle areas (prior to this PR) are also an iffy thing.

comma commented:
So:
1.Boot up torch.
2.Flip breakers few times.
3.Cut and replace cables.
Then I should see the issue, right?

mwerezak commented:
It may be possible to write a unit test that detects cable issues, that could also then be run in-game.

atlantiscze commented:
I was actually considering a generic "alarm" test that checks whether there are any alarms within ~10 minutes since roundstart, but i'm not sure whether unit tests actually support such delay. Alternatively this delay could be emulated by "overclocking" the machinery process to run, let's say, 300 times in a row (300 * 2 = roughly 600s)

mwerezak commented:
Frankly, that sounds like a useless idea.

It would be more useful and more direct to just test that all round-start cables are connected correctly.

We have a precise definition of the problem: Two pieces of cable that connect to each other visually aren't actually connected (don't have the same powernet). Therefore we should test for that.

comma commented:
Okay cutting cables seems to have triggered it on Torch.
Was unable to reproduce with cutting / breakers with https://github.com/Baystation12/Baystation12/pull/15907 so I guess it was caused by up-down cables using power_list instead of same logic as other cables.
If you have a set of steps that can 100% reproduce the issue so I can check with, that'd be nice.
Otherwise https://github.com/Baystation12/Baystation12/pull/15907 and conduits hack are equally 'fixing'.

atlantiscze commented:
Right now the problem isn't that simple. The cables are connected to each other powernet-wise, most of the time, but semi-rarely, something *MAY* (this is the worst) disconnect them. Unless we also make the unit test emulate wire cutting/laying/breaker flipping, the test would probably be more or less useless.

atlantiscze committed:
Finalizes torch changes
- Unit tests should not fail now, unused areas have been removed.
- Maps in few vertical conduit boards into engineering and engineering storage.
- Fixes all off-ship alarms. For construction site it's done by connecting the APCs to small auto-started solar array, for the abandoned outpost it's done by replacing the APC with turned off variant. Also plugs the hole in abandoned outpost to prevent atmosphere alarms from showing up since roundstart.
- Adds new unit test that checks whether all areas have mapped-in ceilings. Replaces /turf/space/initialize(). Should help performance when ceilings are actually mapped in. not a big deal, but applies.
- Fixes telecommunication rooms not having critical flag, and therefore losing power during grid checks and other
- Few other map fixes (few missing wires, etc.)
- Only shuttles with APC have power requirement enabled for now. I'll redo the rest in my bluespace drives PR.

Kasuobes commented:
Give me a few hours and I'll fix those conflicts.

Kasuobes commented:
atlantiscze/Baystation12#4

I'M SORRY T_T

Kasuobes committed:
I'M SORRY (#4)

* Deletes .gitconfig

It doesn't even work there, relevant config is in /.git/config

* Tweaks Ponchos

Re-vamps green and red ponchos. Improves coloring tweaks others to form
to a human shape better.

* Adds Chin Length Bob

A new haircut for those who like bobs, but always hated how short the
one already in game is.

* Shrunk map sizes to 200x200

* NanoMap + changelog

* Fixes slime turboattack

* Gives garden and xenobio more seeds

Garden: 30 of all standard ones minus kudzu and plastellium.
Xenobotany: 30 of all standard ones + 10 random ones.
Also doubles disk amount in the disk box. It fits 14. No reason to have
7.

* Added haywire rounds.

* Less hard numbers for map gen paramenters

* Removes ambrosia

* Uses startswith, more seeds, changelog

* Radiation fix

* Remove outdated golem text

* fixes wrong item path

* Added a TAKE button to adminhelps to try to prevent admin dogpiling.

* Torch manifest and ID console fixes, Exodus tweaks

Fixes service and supply department manifest entries and adds them to the ID computer.
Adds command department to the ID computer.
Splits command department into heads of staff and command support, and adds the support category to the manifest and ID console.

Moved the Exodus cargo squad to Supply
Moved the Exodus chefs, gardeners, bartender and janitor to Service
Moved Internal Affairs Agent to Support

* Spellcheck
organge > orange

Signed-off-by: lorwp <lorwpp@gmail.com>

* correct key size

* Adds .editorconfig

In case people actually like using the web UI.

Kasuobes commented:
Oh shit...

You might need to rebase, but the maps are updated with your changes.

atlantiscze commented:
I have only few minutes now, but i'll do that in the afternoon.

comma commented:
Also conduits aren't needed now that https://github.com/Baystation12/Baystation12/pull/15907 fixes vertical cable connections.

atlantiscze commented:
So, apparently, the fact that other map affecting PRs that were opened **after this one** were merged before this one, everything has broken down. Thank you, developers, for wasting hours of my time.

So find someone else to fix that ton of map bugs, fucked up shuttle power code, half-nonfunctional powernets, broken shield tile acquisition, and bunch of other things i've fixed in this PR. I quit.

Techhead0 commented:
This PR was blocked due to the power conduits. It was requested that you remove them and you refused. You showed no further inclination to remove them even after Z-cable fixes were made. I'm sorry about your wasted time, but I think pinning it entirely on us devs is unfair.

atlantiscze commented:
Smaller PRs aside, two PRs that conflicted each other were made. This one, which resolved a LOT of bugs, including few severe ones, and a second one, that shrunk the maps, in order to slightly improve initialisation time. The shrinking PR could have taken maybe.. twenty minutes to map? This PR took -hours-.

Furthermore, this PR was opened prior to the second one.

I specifically said that these PRs -will- conflict, most likely in an irreversable way. I was ignored, and the other PR was merged, totally fucking up this one even past the point where anything automatic i could come up with failed to work, so i'd have to remap all of those things from scratch again. No thanks. Fix your broken map yourself.

As for the conduits, i directly said that if someone wants to fix the cables in different way, they may do so, but until then, i decided to keep my fix, because in my perspective, it's better to have something that ** actually works ** instead of nothing. This PR was already broken when alternative cable fix was merged, so it was only a small irrelevant nail into the coffin, so don't blame it on that.
